File: src/main/java/insertSort/InsertSort.java
package insertSort;

import java.util.Arrays;

public class InsertSort {
    public static void main(String[] args) {
        int[] A = new int[]{10, 2, 9, 3, 1};
        insertSort(A);
        System.out.println(Arrays.toString(A));
    }

    public static void insertSort(int[] a) {
        for (int i = 0; i < a.length; i++) {
            int key = a[i];
            int prevInx = i - 1;
            while (prevInx >= 0 && a[prevInx] > key) {
                a[prevInx + 1] = a[prevInx];
                prevInx--;
            }
            a[prevInx + 1] = key;
        }
    }
}

File: src/main/java/insertSort/insertSort.kt
fun main() {
    val initList = mutableListOf(10, 5, 9, 6, 7, 8)
    val sortedList = insertSort(initList)
    print(sortedList)
}

fun insertSort(list: MutableList<Int>): List<Int> {
    for ((inx, _) in list.withIndex()) {
        val key = list[inx]
        var prevInx = inx - 1
        while (prevInx >= 0 && list[prevInx] > key) {
            list[prevInx + 1] = list[prevInx]
            prevInx--
        }
        list[prevInx + 1] = key
    }
    return list
}